lib/search.js

```diff
@@ -70,24 +54,69 @@ function search (args, silent, staleness, cb) {
     return s.toLowerCase()
   })
 
-  updateIndex(staleness, opts, searchexclude, filter, function (er, data) {
-    // now data is the list of data that we want to show.
-    // prettify and print it, and then provide the raw
-    // data to the cb.
-    if (er || silent) return cb(er, data)
+  updateIndex(staleness, function (er, allEntriesStream) {
+    if (er) return cb(er)
+    searchInStream(allEntriesStream, filter, args, opts, searchexclude, silent, cb)
+  })
+}
+
+function searchInStream (stream, filter, args, opts, exclude, silent, cb) {
+  log.silly('search', 'searching packages')
+
+  var filterStream = createFilterStream(filter, opts, exclude, silent)
+  var outputStream = npm.config.get('json')
```

Comment:
This was done as part of the refactor. All these options are passed in the `opts` object to the formatter stream.